analyses/molecular-subtyping-LGAT/01-subset-files-for-LGAT.Rmd

```diff
@@ -0,0 +1,227 @@
+
+# Filter to WGS samples
+lgat_wgs_df <- lgat_specimens_df %>%
+  dplyr::filter(experimental_strategy == "WGS") %>%
+  dplyr::select(Kids_First_Biospecimen_ID)
+```
+
+### Subset consensus maf and annotate per subtype based on SNV
+
+```{r}
+# Filter consensus mutation files for LGAT subset
+consensusMutationSubset <- consensusMutation %>%
+  # find lgat samples
+  dplyr::filter(Tumor_Sample_Barcode %in% lgat_wgs_df$Kids_First_Biospecimen_ID) %>%
+  # select tumor sample barcode
+  dplyr::select(Tumor_Sample_Barcode,
+                Hugo_Symbol,
+                HGVSp_Short,
+                DOMAINS,
+                Variant_Classification) %>% 
+  dplyr::distinct() %>%
+  # join other WGS LGAT samples
+  dplyr::full_join(lgat_wgs_df,
+            by = c("Tumor_Sample_Barcode" = "Kids_First_Biospecimen_ID")) %>%
+  dplyr::filter(
+      # get BRAF mutation status
+      # canonical mutations V600E
+      HGVSp_Short %in% snvOI$BRAF_V600E$canonical[!is.na(snvOI$BRAF_V600E$canonical)] &
+        Hugo_Symbol=="BRAF" | # OR
+      # hotspot mutations in p.600 and p.599
+      grepl(BRAF_hotspot,HGVSp_Short) &
+        Hugo_Symbol=="BRAF" &
+        Variant_Classification != "Silent" | # OR
+      # and kinase domain mutation for non-canonical mutation 
+      grepl("PF07714",DOMAINS) & 
+        Hugo_Symbol=="BRAF" & 
+        Variant_Classification != "Silent" | # OR
+    # get NF1 mutation status
+      # all mutations in MAPK genes
+      Hugo_Symbol %in% snvOI$NF1$gene & 
+        Variant_Classification %in% c("Missense_Mutation","Nonsense_Mutation") |
+    # other MAPK mutation status
+      # all mutations in MAPK genes
+      Hugo_Symbol %in% snvOI$MAPK$gene | # OR
+    # RTK mutation status
+      # all mutations in RTK genes
+      Hugo_Symbol %in% snvOI$RTK$gene | # OR
+    # get FGFR mutation status
+      # canonical mutations
+      HGVSp_Short %in% snvOI$FGFR$canonical[!is.na(snvOI$FGFR$canonical)] &
+        Hugo_Symbol=="FGFR1" | # OR
+      # hotspot mutations 
+      grepl(FGFR_hotspot,HGVSp_Short) &
+        Hugo_Symbol=="FGFR1" | # OR
+    # get IDH mutation status
+      # hostspot mutations
+      grepl(IDH_hotspot,HGVSp_Short) & 
+        Hugo_Symbol %in% snvOI$IDH$gene | # OR
+    # get H3F3A mutation status
+      # canonical mutations
+      HGVSp_Short %in% snvOI$H3F3A$canonical & Hugo_Symbol %in% "H3F3A" | # OR
+    # get HIST1H3B mutation status
+      # canonical mutations
+      HGVSp_Short %in% snvOI$HIST1H3B$canonical & Hugo_Symbol %in% "HIST1H3B" | # OR
+    # get HIST1H3C mutation status
+      # canonical mutations
+      HGVSp_Short %in% snvOI$HIST1H3C$canonical & Hugo_Symbol %in% "HIST1H3C" 
+  ) %>%
+  dplyr::distinct()
+
+consensusMutationSubset
+```
+
+
+### Gather status if multiple
+
+```{r}
+
+consensusMutationSubset <- lgat_wgs_df %>%
+  left_join(consensusMutationSubset,by=c("Kids_First_Biospecimen_ID"="Tumor_Sample_Barcode")) %>%
```

Comment:
```suggestion
consensusMutationSubset <-   lgat_bsids_dna %>%
as_tibble_col(column_name = "Tumor_Sample_Barcode") %>%
left_join(consensusMutationSubset) %>%
```
I think this should work to simplify here if you make `lgat_bsids_dna` as a list above